test.py

```diff
@@ -0,0 +1,65 @@
+import os
+import pytest
+import subprocess
+import testinfra
+
+@pytest.fixture(scope='session')
+def host(request):
+    subprocess.check_call(['docker', 'build', '-t', 'radicale', '.'])
+    docker_id = subprocess.check_output(['docker', 'run', '-d', 'radicale']).decode().strip()
+    
+    yield testinfra.get_host("docker://" + docker_id)
+    
+    # teardown
+    subprocess.check_call(['docker', 'rm', '-f', docker_id])
+
+def test_system(host):
+    assert host.system_info.distribution == 'alpine'
+    assert host.system_info.release == '3.7.0'
+
+def test_entrypoint(host):
+    entrypoint = '/usr/local/bin/docker-entrypoint.sh'
+    assert host.file(entrypoint).exists
+    assert oct(host.file(entrypoint).mode) == '0o555'
+
+def test_process(host):
+    process = host.process.get(comm='radicale')
+    assert process.pid == 1
+    assert process.user == 'radicale'
+    assert '/usr/bin/radicale --config /config/config' in process.args
+
+def test_port(host):
+    assert host.socket("tcp://0.0.0.0:5232").is_listening
```

Comment:
This test case could set a custom port and try if radicale is bound to it

Reply:
It would mean changing the default config prior to building the container, which would then differ to the actual image we build. 

I like the concept of testing a random port from a testing radicale perspective as that should work. Whereas this test it to check the image we build, runs by default on port 5232.